src/Columns/ColumnConst.h

```diff
@@ -253,7 +254,22 @@ class ColumnConst final : public COWHelper<IColumn, ColumnConst>
     const IColumn & getDataColumn() const { return *data; }
     const ColumnPtr & getDataColumnPtr() const { return data; }
 
-    Field getField() const { return getDataColumn()[0]; }
+    Field getField() const 
+    { 
+        auto & col = getDataColumn();
+        auto * array_col = typeid_cast<ColumnArray *>(const_cast<IColumn*>(&col));
+        //if it can successfully down-convert to ColumnArray 
+        if (array_col)
+        {
+            auto field_size = array_col->getOffsets()[0];
+            if (field_size > max_array_size_as_field)
+            {
+                Field zero = 0;
+                return zero;
+            }
+        }
+        return col[0];
+    }
```

Comment:
```
    Field getField() const
    {
        auto col = getDataColumnPtr();
        Field field;
        const auto * array_col = typeid_cast<const ColumnArray *>(col.get());
        //if it can successfully down-convert to ColumnArray
        if (array_col)
        {
            auto field_size = array_col->getOffsets()[0];
            if (field_size > max_array_size_as_field)
            {
                field = 0;
            }
        }
        else
            col->get(0, field);

        return field;
    }
```
Maybe replace getDataColumnPtr and appropriate methods later.